src/adapters/module-service.ts

```diff
@@ -0,0 +1,88 @@
+import {
+  type EdgeSpecAdapter,
+  type EdgeSpecRouteBundle,
+  handleRequestWithEdgeSpec,
+} from "src/types/edge-spec"
+import { EdgeSpecRequest, EdgeSpecRouteFn } from "src/types/web-handler"
+
+export interface ModuleServiceOptions {
+  routeParam?: string
+  handleRouteParamNotFound?: EdgeSpecRouteFn
+  allowMatchingOnAnyCatchAllRouteParam?: boolean
+}
+
+export type ModuleService = (options?: ModuleServiceOptions) => EdgeSpecRouteFn
+
+export const createModuleService: EdgeSpecAdapter<[], ModuleService> = (
```

Comment:
🐧 I kinda like 
```suggestion
export const createEmbeddedService: EdgeSpecAdapter<[], EmbeddedService> = (
```

but don't want to bikeshed too much

Reply:
@seveibar 

will deal with this in follow-up discussions/PRs/RFCs